fairlearn/preprocessing/_correlation_remover.py

```diff
@@ -113,11 +116,26 @@ def transform(self, X):
         X_filtered = np.atleast_2d(X_filtered)
         return self.alpha * X_filtered + (1 - self.alpha) * X_use
 
+    def _check_sensitive_features_in_X(self, X) -> None:
+        """Check if the sensitive features are in X."""
+        if isinstance(X, pd.DataFrame):
+            missing_columns = [c for c in self.sensitive_feature_ids if c not in X.columns]
+        else:
+            if X.ndim == 1:
+                return
+            missing_columns = [i for i in self.sensitive_feature_ids if i not in range(X.shape[1])]
+
+        if len(missing_columns) > 0:
+            raise ValueError(f"Columns {missing_columns} not found in the input data.")
+
     def _more_tags(self):
         return {
             "_xfail_checks": {
                 "check_transformer_data_not_an_array": (
                     "this estimator only accepts pandas dataframes or numpy ndarray as input."
                 ),
+                "check_estimators_empty_data_messages": (
+                    "this estimator raises on missing sensitive features in data."
+                ),
```

Comment:
The `Estimator` can handle empty data fine. But we still throw an exception in case the data is empty and the `sensitive_feature_ids` are not.